### ISSUE TITLE
The preview stops when I switch between layouts.

### ISSUE THREAD

**Issue by ERAgames777** [Scirra/Construct-3-bugs#3062]
The preview stops when I switch between layouts in my project. This happens exclusively in c3runtime, when I use c2runtime, the project preview works without stopping. The stop does not happen immediately when I turn 1-2 times between layouts and return back to them. I looked through the list of errors in the chrome console and this is what I see:

```
include.preload.js:335 [Deprecation] Element.createShadowRoot is deprecated and will be removed in M73, around March 2019. Please use Element.attachShadow instead. See https://www.chromestatus.com/features/4507242028072960 for more details.
createShadowTree @ include.preload.js:335
bootPreview.js:14 Registered service worker on https://preview.construct.net/local/
bootPreview.js:8 Unexpected message type 'boot-ready'
OnBootMessage @ bootPreview.js:8
domSide.js:1 [C3 IAP] Initialising
runtime.js:1 [C3 runtime] Hosted in DOM, rendering with WebGL 2 [ANGLE (Intel(R) HD Graphics Family Direct3D11 vs_5_0 ps_5_0)]
9ddff747-518b-467b-9869-f78baebb105d:9 BHT_Smart_RandomInstance:constructor-OUT
9ddff747-518b-467b-9869-f78baebb105d:49 Regenerate()
9ddff747-518b-467b-9869-f78baebb105d:39 BHT_Smart_RandomInstance:constructor-OUT
domSide.js:1 Rejection from 'advert' handler 'Configure': Error: advert plugin not loaded
    at a._CallMethod (domSide.js:1)
    at domSide.js:1
    at e._OnEventFromRuntime (domSide.js:1)
    at e._OnMessageFromRuntime (domSide.js:1)
    at MessagePort._messageChannelPort.onmessage (domSide.js:1)
(anonymous) @ domSide.js:1
instance.js:1 [C3 advert] Event (configuration failed Error (Error: advert plugin not loaded) Type () Amount (0))
domSide.js:1 Rejection from 'advert' handler 'CreateInterstitialAdvert': Error: advert plugin not loaded
    at a._CallMethod (domSide.js:1)
    at domSide.js:1
    at e._OnEventFromRuntime (domSide.js:1)
    at e._OnMessageFromRuntime (domSide.js:1)
    at MessagePort._messageChannelPort.onmessage (domSide.js:1)
(anonymous) @ domSide.js:1
instance.js:1 [C3 advert] Event (failed to create interstitial Error (Error: advert plugin not loaded) Type () Amount (0))
domSide.js:1 Rejection from 'advert' handler 'CreateInterstitialAdvert': Error: advert plugin not loaded
    at a._CallMethod (domSide.js:1)
    at domSide.js:1
    at e._OnEventFromRuntime (domSide.js:1)
    at e._OnMessageFromRuntime (domSide.js:1)
    at MessagePort._messageChannelPort.onmessage (domSide.js:1)
(anonymous) @ domSide.js:1
domSide.js:1 Rejection from 'advert' handler 'CreateVideoAdvert': Error: advert plugin not loaded
    at a._CallMethod (domSide.js:1)
    at domSide.js:1
    at e._OnEventFromRuntime (domSide.js:1)
    at e._OnMessageFromRuntime (domSide.js:1)
    at MessagePort._messageChannelPort.onmessage (domSide.js:1)
(anonymous) @ domSide.js:1
instance.js:1 [C3 advert] Event (failed to create interstitial Error (Error: advert plugin not loaded) Type () Amount (0))
instance.js:1 [C3 advert] Event (failed to create video Error (Error: advert plugin not loaded) Type () Amount (0))
domSide.js:1 Rejection from 'advert' handler 'CreateInterstitialAdvert': Error: advert plugin not loaded
    at a._CallMethod (domSide.js:1)
    at domSide.js:1
    at e._OnEventFromRuntime (domSide.js:1)
    at e._OnMessageFromRuntime (domSide.js:1)
    at MessagePort._messageChannelPort.onmessage (domSide.js:1)
(anonymous) @ domSide.js:1
domSide.js:1 Rejection from 'advert' handler 'CreateVideoAdvert': Error: advert plugin not loaded
    at a._CallMethod (domSide.js:1)
    at domSide.js:1
    at e._OnEventFromRuntime (domSide.js:1)
    at e._OnMessageFromRuntime (domSide.js:1)
    at MessagePort._messageChannelPort.onmessage (domSide.js:1)
(anonymous) @ domSide.js:1
instance.js:1 [C3 advert] Event (failed to create interstitial Error (Error: advert plugin not loaded) Type () Amount (0))
instance.js:1 [C3 advert] Event (failed to create video Error (Error: advert plugin not loaded) Type () Amount (0))
domSide.js:1 Rejection from 'advert' handler 'CreateInterstitialAdvert': Error: advert plugin not loaded
    at a._CallMethod (domSide.js:1)
    at domSide.js:1
    at e._OnEventFromRuntime (domSide.js:1)
    at e._OnMessageFromRuntime (domSide.js:1)
    at MessagePort._messageChannelPort.onmessage (domSide.js:1)
(anonymous) @ domSide.js:1
domSide.js:1 Rejection from 'advert' handler 'CreateVideoAdvert': Error: advert plugin not loaded
    at a._CallMethod (domSide.js:1)
    at domSide.js:1
    at e._OnEventFromRuntime (domSide.js:1)
    at e._OnMessageFromRuntime (domSide.js:1)
    at MessagePort._messageChannelPort.onmessage (domSide.js:1)
(anonymous) @ domSide.js:1
instance.js:1 [C3 advert] Event (failed to create interstitial Error (Error: advert plugin not loaded) Type () Amount (0))
instance.js:1 [C3 advert] Event (failed to create video Error (Error: advert plugin not loaded) Type () Amount (0))
domSide.js:1 Rejection from 'advert' handler 'CreateInterstitialAdvert': Error: advert plugin not loaded
    at a._CallMethod (domSide.js:1)
    at domSide.js:1
    at e._OnEventFromRuntime (domSide.js:1)
    at e._OnMessageFromRuntime (domSide.js:1)
    at MessagePort._messageChannelPort.onmessage (domSide.js:1)
(anonymous) @ domSide.js:1
domSide.js:1 Rejection from 'advert' handler 'CreateVideoAdvert': Error: advert plugin not loaded
    at a._CallMethod (domSide.js:1)
    at domSide.js:1
    at e._OnEventFromRuntime (domSide.js:1)
    at e._OnMessageFromRuntime (domSide.js:1)
    at MessagePort._messageChannelPort.onmessage (domSide.js:1)
(anonymous) @ domSide.js:1
instance.js:1 [C3 advert] Event (failed to create interstitial Error (Error: advert plugin not loaded) Type () Amount (0))
instance.js:1 [C3 advert] Event (failed to create video Error (Error: advert plugin not loaded) Type () Amount (0))
domSide.js:1 Rejection from 'advert' handler 'CreateInterstitialAdvert': Error: advert plugin not loaded
    at a._CallMethod (domSide.js:1)
    at domSide.js:1
    at e._OnEventFromRuntime (domSide.js:1)
    at e._OnMessageFromRuntime (domSide.js:1)
    at MessagePort._messageChannelPort.onmessage (domSide.js:1)
(anonymous) @ domSide.js:1
domSide.js:1 Rejection from 'advert' handler 'CreateVideoAdvert': Error: advert plugin not loaded
    at a._CallMethod (domSide.js:1)
    at domSide.js:1
    at e._OnEventFromRuntime (domSide.js:1)
    at e._OnMessageFromRuntime (domSide.js:1)
    at MessagePort._messageChannelPort.onmessage (domSide.js:1)
(anonymous) @ domSide.js:1
instance.js:1 [C3 advert] Event (failed to create interstitial Error (Error: advert plugin not loaded) Type () Amount (0))
instance.js:1 [C3 advert] Event (failed to create video Error (Error: advert plugin not loaded) Type () Amount (0))
domSide.js:1 Rejection from 'advert' handler 'CreateInterstitialAdvert': Error: advert plugin not loaded
    at a._CallMethod (domSide.js:1)
    at domSide.js:1
    at e._OnEventFromRuntime (domSide.js:1)
    at e._OnMessageFromRuntime (domSide.js:1)
    at MessagePort._messageChannelPort.onmessage (domSide.js:1)
(anonymous) @ domSide.js:1
domSide.js:1 Rejection from 'advert' handler 'CreateVideoAdvert': Error: advert plugin not loaded
    at a._CallMethod (domSide.js:1)
    at domSide.js:1
    at e._OnEventFromRuntime (domSide.js:1)
    at e._OnMessageFromRuntime (domSide.js:1)
    at MessagePort._messageChannelPort.onmessage (domSide.js:1)
(anonymous) @ domSide.js:1
instance.js:1 [C3 advert] Event (failed to create interstitial Error (Error: advert plugin not loaded) Type () Amount (0))
instance.js:1 [C3 advert] Event (failed to create video Error (Error: advert plugin not loaded) Type () Amount (0))
imageInfo.js:1 Uncaught (in promise) Error: already loaded texture
    at C3.ImageInfo.LoadStaticTexture (imageInfo.js:1)
    at animationInfo.js:1
    at Array.map (<anonymous>)
    at C3.AnimationInfo.LoadAllTextures (animationInfo.js:1)
    at type.js:1
    at Array.map (<anonymous>)
    at C3.Plugins.Sprite.Type.LoadTextures (type.js:1)
    at C3.ObjectClass.LoadTextures (objectClass.js:1)
    at C3.Layout._Load (layout.js:1)
    at C3.Runtime._DoChangeLayout (runtime.js:1)
runtime.js:1 [Construct 3] Suspending
runtime.js:1 [Construct 3] Resuming
runtime.js:1 Uncaught (in promise) TypeError: Cannot read property '_StopRunning' of null
    at C3.Runtime._DoChangeLayout (runtime.js:1)
    at C3.Runtime._MaybeChangeLayout (runtime.js:1)
    at C3.Runtime.Step (runtime.js:1)
    at async C3.Runtime.Tick (runtime.js:1)

```

**Comment by shortercode** [comment]
Most of this appears to be noise, the Advert plugin tends to be a bit wordy if it's used out of it's normal environment. The section at the end:

```
imageInfo.js:1 Uncaught (in promise) Error: already loaded texture
    at C3.ImageInfo.LoadStaticTexture (imageInfo.js:1)
    at animationInfo.js:1
    at Array.map (<anonymous>)
    at C3.AnimationInfo.LoadAllTextures (animationInfo.js:1)
    at type.js:1
    at Array.map (<anonymous>)
    at C3.Plugins.Sprite.Type.LoadTextures (type.js:1)
    at C3.ObjectClass.LoadTextures (objectClass.js:1)
    at C3.Layout._Load (layout.js:1)
    at C3.Runtime._DoChangeLayout (runtime.js:1)
runtime.js:1 [Construct 3] Suspending
runtime.js:1 [Construct 3] Resuming
runtime.js:1 Uncaught (in promise) TypeError: Cannot read property '_StopRunning' of null
    at C3.Runtime._DoChangeLayout (runtime.js:1)
    at C3.Runtime._MaybeChangeLayout (runtime.js:1)
    at C3.Runtime.Step (runtime.js:1)
    at async C3.Runtime.Tick (runtime.js:1)
```

Looks more interesting. An example project and a replication would be appreciated.

**Comment by ERAgames777** [comment]
I can not provide a project, as it is a finished game. I tried to implement this stop in an empty project, but it did not work out.

**Comment by ERAgames777** [comment]
@shortercode 
I can send you screenshots from the finished project. Perhaps you will understand what is happening and the video as the game stops when I switch between the layouts. If you need that answer me.

**Comment by shortercode** [comment]
I'm only really able to deal with the triage on this bug. Texture loading and the core graphics components in the runtime is the territory of @AshleyScirra, so he's the one to talk to. Most of the code around this area is quite complicated. I think the bare minimum we need here is project that reproduces the failure.

**Comment by AshleyScirra** [comment]
The real problem appears to be a crash changing layout:

```
TypeError: Cannot read property '_StopRunning' of null
    at C3.Runtime._DoChangeLayout (runtime.js:1)
    at C3.Runtime._MaybeChangeLayout (runtime.js:1)
    at C3.Runtime.Step (runtime.js:1)
    at async C3.Runtime.Tick (runtime.js:1)
```

Everything else looks like a knock-on effect from that.

Unfortunately from this I have no idea why it might be going wrong. Please follow all the bug report guidelines - we need all that information to be able to help.
